modules/engines/engine-xsodata/src/test/java/com/sap/xsk/xsodata/ds/handler/XSKOData2EventHandlerTest.java

```diff
@@ -0,0 +1,347 @@
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.xsodata.ds.handler;
+
+import com.google.gson.internal.LinkedTreeMap;
+import com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils;
+import org.apache.olingo.odata2.api.commons.HttpStatusCodes;
+import org.apache.olingo.odata2.api.edm.EdmException;
+import org.apache.olingo.odata2.api.edm.EdmType;
+import org.apache.olingo.odata2.api.ep.EntityProviderException;
+import org.apache.olingo.odata2.api.ep.entry.ODataEntry;
+import org.apache.olingo.odata2.api.processor.ODataResponse;
+import org.apache.olingo.odata2.api.uri.UriInfo;
+import org.apache.olingo.odata2.core.ep.BasicEntityProvider;
+import org.eclipse.dirigible.commons.api.helpers.GsonHelper;
+import org.eclipse.dirigible.commons.config.StaticObjects;
+import org.eclipse.dirigible.engine.odata2.api.ODataException;
+import org.eclipse.dirigible.engine.odata2.definition.ODataHandlerDefinition;
+import org.eclipse.dirigible.engine.odata2.service.ODataCoreService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.sql.DataSource;
+import java.io.InputStream;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.AFTER_TABLE_NAME;
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.BEFORE_DELETE_ENTITY_TABLE_NAME;
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.BEFORE_TABLE_NAME;
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.BEFORE_UPDATE_ENTITY_TABLE_NAME;
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.CONNECTION;
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.ENTRY_MAP;
+import static com.sap.xsk.xsodata.utils.XSKOData2EventHandlerUtils.ON_CREATE_ENTITY_TABLE_NAME;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+
+@RunWith(MockitoJUnitRunner.class)
+public class XSKOData2EventHandlerTest {
+
+  @Mock
+  private EdmType edmType;
+
+  @Mock
+  private UriInfo uriInfo;
+
+  @Mock
+  private ODataCoreService odataCoreService;
+
+  @Mock
+  private DataSource dataSource;
+
+  @Mock
+  private ODataEntry entry;
+
+  @Mock
+  private InputStream inputStream;
+
+  private static MockedStatic<XSKOData2EventHandlerUtils> handlerUtils;
+
```

Comment:
why don't you use standard class instance, instead of mocking static objects? Static objects/method should be created in rare cases where there is no other way to achieve something. Research/read about the Dependancy injection principle.